lib/item.rb

```diff
@@ -0,0 +1,59 @@
+require 'httparty'
+require "pry"
+
+class Item
+  attr_reader :nypl_source, :id
+  attr_accessor :item_type_code, :location_code
+
+  def initialize(nypl_source, id)
+    @nypl_source = nypl_source
+    @id = id
+  end
+
+  def is_research?
+    get_platform_api_data
+
+    result = is_partner? || item_type_is_research? || location_is_only_research?
+
+    log_data = {is_partner: is_partner?}
+
+    if !is_partner?
+      log_data[:item_type_is_research] = item_type_is_research?
+      log_data[:location_is_only_research] = location_is_only_research?
+    end
+
+    $logger.debug "Evaluating is-research for #{nypl_source} #{id}: #{result}", log_data
+
+    return result
+  end
+
+  private
+    def is_partner?
+      nypl_source == "recap-cul" || nypl_source == "recap-pul"
+    end
+
+    def item_type_is_research?
+      item_collection_type = $nypl_core.by_catalog_item_type[item_type_code]
+      return item_collection_type["collectionType"].include?("Research")
```

Comment:
Might want to think about what happens if item_collection_type["collectionType"] isn't a string